packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -147,10 +158,18 @@ export class CdkToolkit {
         const currentTemplate = templateWithNames.deployedTemplate;
         const nestedStackCount = templateWithNames.nestedStackCount;
 
+        const changeSet = options.changeSet ? await prepareAndCreateChangeSet({
```

Comment:
Just for safety's sake, I think I'd like us to catch any permission errors here, print a warning, and fall back to the "less accurate" diff.

Otherwise, I'm concerned this mandatory new API call is going to break for some people somewhere.